buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy

```diff
@@ -49,25 +55,38 @@ public class NoticeTask extends DefaultTask {
         dependencies.add(project)
     }
 
+    /** Add notices from an extra directory. */
+    public void extraLicensesDir(File extraLicensesDir) {
+        extraLicensesDirs.add(extraLicensesDir)
+    }
+
     @TaskAction
     public void generateNotice() {
         StringBuilder output = new StringBuilder()
         output.append(inputFile.getText('UTF-8'))
         output.append('\n\n')
-        Set<String> seen = new HashSet<>()
+        Map<String, File> seen = new TreeMap<>()
```

Comment:
Having the key be the name of the file without any path means the contents of the resulting NOTICE.txt is sorted by the names of the 3rd party components.